sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -623,6 +632,22 @@ public void Dispose()
                 linkSettings.AddProperty(AmqpProperty.Timeout, (uint)timeout.CalculateRemaining(stopWatch.GetElapsedTime()).TotalMilliseconds);
                 linkSettings.AddProperty(AmqpProperty.EntityType, (int)AmqpProperty.Entity.EventHub);
 
+                if ((features & TransportProducerFeatures.IdempotentPublishing) != 0)
+                {
+                    linkSettings.DesiredCapabilities ??= new Multiple<AmqpSymbol>();
+                    linkSettings.DesiredCapabilities.Add(AmqpProperty.EnableIdempotentPublishing);
+                }
+
+                // These values must either all be specified or none.  It is valid for an individual value to be null to signal the
+                // service to supply the value, so long as the key is present.
+
+                if ((options.ProducerGroupId.HasValue) || (options.OwnerLevel.HasValue) || (options.StartingSequenceNumber.HasValue))
```

Comment:
based on the comment "_These values must either all be specified or none_" I would have expected this to be an `&&` expression. But perhaps I'm confused as to the meaning of "_so long as the key is present_"

Reply:
It's valid for the options to only specify a partial set of values, but we need to ensure that there are entries in the map for all.  Basically, if any value is set, then all of the keys must be in the AMQP map, even if the value is `null`.

Reply:
I'll try to add a bit of clarifying context in the next set of changes.